LanguageFeatures/Patterns/constant_A06_t01.dart

```diff
@@ -61,6 +61,10 @@ main() {
     case C():
```

Comment:
If we maintain that this `<caseHead>` has a `<constantPattern>` then this would need to be `case const C():` (or something else, but it can't just be `case C():` because that's a syntax error, not a constant pattern). I think it would be useful to have `case const C()` because it uses the alternative `<constObjectExpression>` in `<constantPattern>`.

Another thing we ought to have is a string literal which isn't constant (like `'x is $x!'`).